File: frontend/src/lib/runtime-config.ts
import { apiClient } from './api-client';

export interface RuntimeConfig {
  product_name: string;
  version: string;
  features: {
    llm_enabled: boolean;
    smtp_enabled: boolean;
    imap_enabled: boolean;
    dev_header_auth_enabled: boolean;
    manual_bearer_login_enabled: boolean;
  };
}

let runtimeConfigPromise: Promise<RuntimeConfig> | null = null;

export function getRuntimeConfig() {
  if (!runtimeConfigPromise) {
    runtimeConfigPromise = apiClient.get<RuntimeConfig>('/api/runtime-config').then((config) => {
      apiClient.setDevHeaderAuthEnabled(config.features.dev_header_auth_enabled);
      return config;
    });
  }
  return runtimeConfigPromise;
}

export function resetRuntimeConfigCache() {
  runtimeConfigPromise = null;
}

File: backend/api/runtime_config.py
from fastapi import APIRouter
from pydantic import BaseModel

from core.version import get_release_version

router = APIRouter(prefix="/api/runtime-config", tags=["runtime-config"])


class RuntimeConfigResponse(BaseModel):
    product_name: str
    version: str
    features: dict[str, bool]


@router.get("", response_model=RuntimeConfigResponse)
async def get_runtime_config():
    # Return basic non-secret configuration
    return RuntimeConfigResponse(
        product_name="Naruon",
        version=get_release_version(),
        features={
            "llm_enabled": True,
            "smtp_enabled": True,
            "imap_enabled": True,
        },
    )
